### PR TITLE
feat:add metering module

### PR DESCRIPTION
**Pull request by xiao-jay**
#1514
The billing module measures the resources used by infra and user ns, and then deducts the user's account balance.

### CODE REVIEW

File: controllers/metering/config/crd/bases/metering.sealos.io_meterings.yaml [hunk 1]
```diff
@@ -15,7 +15,18 @@ spec:
     singular: metering
   scope: Namespaced
   versions:
-  - name: v1
+  - additionalPrinterColumns:
+    - jsonPath: .spec.owner
+      name: owner
+      type: string
+    - description: ' The last two digits are decimals ,100 = 1$'
```
Comment (fanux) [inline controllers/metering/config/crd/bases/metering.sealos.io_meterings.yaml:22]: $ -> ¥

File: controllers/metering/controllers/metering_controller.go [hunk 2]
```diff
@@ -14,23 +14,48 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// 找到owner的metering-----之后再说
```
Comment (fanux) [inline controllers/metering/controllers/metering_controller.go:17]: English

File: controllers/metering/controllers/metering_controller.go [hunk 3]
```diff
@@ -39,23 +64,217 @@ type MeteringReconciler struct {
...
+
+// metering belong to account，resourceQuota belong to metering
+
+func (r *MeteringReconcile) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Logger = log.FromContext(ctx)
+	var metering meteringv1.Metering
+	r.Logger.Info("enter reconcile", "name: ", req.Name, "namespace: ", req.Namespace)
+
+	// if create a user namespace ,will enter this reconcile
+	if req.Name != SealosSystemNamespace {
```
Comment (fanux) [inline controllers/metering/controllers/metering_controller.go:91]: Using req.Kind

File: controllers/metering/config/crd/bases/metering.sealos.io_meterings.yaml (hunk 1 above)
Comment (xiao-jay) [reply controllers/metering/config/crd/bases/metering.sealos.io_meterings.yaml]: 👌

File: controllers/metering/controllers/metering_controller.go (hunk 2 above)
Comment (xiao-jay) [reply controllers/metering/controllers/metering_controller.go]: ok

**Comment by xiao-jay** [comment]
- [x] 验证infra扣费模块是否正常扣费
- [ ] 计量用户ns是否能够正常扣费
- [ ] 用户是否能够自己操作resourcequota

**Comment by fanux** [comment]
add USER in your Dockerfile

File: controllers/metering/controllers/metering_controller.go (hunk 3 above)
Comment (xiao-jay) [reply controllers/metering/controllers/metering_controller.go:88]: req.Kind is not found

**Comment by fanux** [comment]
The controller pod must run as nonroot, like:
```
    # ok: run-as-non-root
    - name: haproxy
      image: haproxy
      securityContext:
        runAsNonRoot: true
```

**Review by fanux — APPROVED** [review]

**Pull request merged**